components/dashboard/src/settings/SSHKeys.tsx

```diff
@@ -0,0 +1,258 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { useContext, useEffect, useState } from "react";
+import { PageWithSubMenu } from "../components/PageWithSubMenu";
+import getSettingsMenu from "./settings-menu";
+import { PaymentContext } from "../payment-context";
+import Modal from "../components/Modal";
+import Alert from "../components/Alert";
+import { Item, ItemField, ItemFieldContextMenu } from "../components/ItemsList";
+import ConfirmationModal from "../components/ConfirmationModal";
+import { SSHPublicKeyValue, UserSSHPublicKeyValue } from "@gitpod/gitpod-protocol";
+import { getGitpodService } from "../service/service";
+import moment from "moment";
+
+interface AddModalProps {
+    value: SSHPublicKeyValue;
+    onClose: () => void;
+    onSave: () => void;
+}
+
+interface DeleteModalProps {
+    value: UserSSHPublicKeyValue;
+    onConfirm: () => void;
+    onClose: () => void;
+}
+
+export function AddSSHKeyModal(props: AddModalProps) {
+    const [errorMsg, setErrorMsg] = useState("");
+
+    const [value, setValue] = useState({ ...props.value });
+    const update = (pev: Partial<SSHPublicKeyValue>) => {
+        setValue({ ...value, ...pev });
+        setErrorMsg("");
+    };
+
+    useEffect(() => {
+        setValue({ ...props.value });
+        setErrorMsg("");
+    }, [props.value]);
+
+    const save = async () => {
+        const tmp = SSHPublicKeyValue.validate(value);
+        if (tmp) {
+            setErrorMsg(tmp);
+            return false;
+        }
+        try {
+            await getGitpodService().server.addSSHPublicKey(value);
+        } catch (e) {
+            setErrorMsg(e.message.replace("Request addSSHPublicKey failed with message: ", ""));
+            return false;
+        }
+        props.onClose();
+        props.onSave();
+        return true;
+    };
+
+    return (
+        <Modal
+            title="New SSH Key"
+            buttons={
+                <button className="ml-2" onClick={save}>
+                    Add SSH Key
+                </button>
+            }
+            visible={true}
+            onClose={props.onClose}
+            onEnter={save}
+        >
+            <>
+                {errorMsg.length > 0 && (
+                    <Alert type="error" className="mb-2">
+                        {errorMsg}
+                    </Alert>
+                )}
+            </>
+            <div className="text-gray-500 dark:text-gray-400 text-md">
+                Add an SSH key for secure access workspaces via SSH.{" "}
+                <a href="/docs/configure/ssh" target="gitpod-ssh-doc" className="gp-link">
+                    Learn more
+                </a>
```

Comment:
**thought**: Do we need this help link here? Asking because we also link below to the same page.